src/Components/Components/src/ComponentBase.cs

```diff
@@ -41,6 +43,16 @@ public ComponentBase()
         };
     }
 
+    /// <summary>
+    /// Gets the <see cref="ComponentPlatform"/> the component is running on.
+    /// </summary>
+    protected ComponentPlatform Platform => _renderHandle.Platform;
+
+    /// <summary>
+    /// Gets the <see cref="IComponentRenderMode"/> the component is rendering or will render in when it becomes interactive.
+    /// </summary>
+    protected IComponentRenderMode? InteractiveRenderMode => _interactiveRenderMode ??= _renderHandle.GetInteractiveRenderMode();
```

Comment:
Could we just call it RenderMode? The return value is valid even if the component isn't interactive, plus we could add other noninteractive rendermodes in the future.

Also it would be good to change the caching rule so it caches even if the value is null.

Reply:
The reason I wanted to add interactive in there is because I'm worried of people writing

```
if(RenderMode is InteractiveServer) 
{
  `JSRuntime.InvokeAsync(...)`
}
```

So I wanted to give it a name that gave a hint as to what this meant/covered and didn't want to call it NonPrerendered render mode.

Reply:
Fair enough, let's go with RenderMode then.

Reply:
I had to do GetRenderMode() instead because RenderMode is a type and that will cause all sort of symbol resolution issues